### PR TITLE
Add new package: influxdb

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/influxdb/package.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        key = "{0}_{1}".format(platform.system(), platform.machine())
+        pkg = packages.get(key)
+        if pkg:
+            version(ver, sha256=pkg[0], url=pkg[1])
+
+    def setup_run_environment(self, env):
+        env.prepend_path('PATH', self.prefix.usr.bin)
+
+    def install(self, spec, prefix):
+        install_tree('.', prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/influxdb/package.py]: What if we instead install `usr` to `prefix`? Will everything be in the right location then?

Comment (darmac) [reply var/spack/repos/builtin/packages/influxdb/package.py]:
```
# spack cd influxdb
# ls
etc  usr  var
```

I don't understand how to do that.
Is that anything wrong we copy `etc` , `usr`, `var` to `prefix`?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/influxdb/package.py]: Most packages expect executables to be in `prefix.bin`, libraries to be in `prefix.lib`, etc.

Could you do something like:
```python
install_tree('usr', prefix)
install_tree('etc', prefix.etc)
install_tree('var', prefix.var)
```
Will that produce the desired result?

Comment (darmac) [reply var/spack/repos/builtin/packages/influxdb/package.py]: I tried this in https://github.com/spack/spack/pull/17909/commits/3886ad5d753cdce2157a78d056ce648351beb8fa
and looks good.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**